nautobot/dcim/tests/test_api.py

```diff
@@ -2769,3 +2771,72 @@ def setUpTestData(cls):
                 "device_type": device_types[3].pk,
             },
         ]
+
+
+class ControllerTestCase(APIViewTestCases.APIViewTestCase):
+    model = Controller
+
+    @classmethod
+    def setUpTestData(cls):
+        statuses = Status.objects.get_for_model(Controller)
+        roles = Role.objects.get_for_model(Controller)
+        platforms = Platform.objects.all()
+        locations = Location.objects.all()
```

Comment:
Should be `Location.objects.get_for_model(Controller).all()`

Actually, are we missing validation on `Controller` model to make sure it's only allocated to a LocationType that permits Controller as a content-type?

Additionally, probably need to add Controller content-type to `LocationTypeFactory.content_types`...

Reply:
Fixed, added to factory, added validation and unittest.